tests/bookmarks.test.js

```diff
@@ -0,0 +1,85 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../server/app';
+
+chai.use(chaiHttp);
+let userToken;
+describe('TEST BOOKMARK ROUTE', () => {
+  it('logs in a user', done => {
+    chai
+      .request(app)
+      .post('/api/v1/auth/login')
+      .send({
+        email: 'vic3coorp@gmail.com',
+        password: '12345678',
+      })
+      .end((err, res) => {
+        const { token } = res.body.user;
+        expect(res.status).to.equal(200);
+        userToken = token;
+        done();
+      });
+  });
+
+  it('It should return a 201 if a user successfully bookmarks an article', done => {
+    chai
+      .request(app)
+      .post('/api/v1/bookmarks/7139d3af-b8b4-44f6-a49f-9305791700f4')
+      .set('authorization', userToken)
+      .end((err, res) => {
+        expect(res.status).to.equal(201);
+        expect(res.body.message).to.be.a('string');
+        expect(res.body.message).to.equal('Successfuly added Bookmark');
+        done();
+      });
+  });
+
+  it('It should return a 200 if a user succesfully removes a bookmark', done => {
+    chai
+      .request(app)
+      .post('/api/v1/bookmarks/7139d3af-b8b4-44f6-a49f-9305791700f4')
+      .set('authorization', userToken)
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body.message).to.be.a('string');
+        expect(res.body.message).to.equal('Successfully removed Bookmark');
+        done();
+      });
+  });
+  it('It should return a 403 error for an invalid token', done => {
+    chai
+      .request(app)
+      .post('/api/v1/bookmarks/7139d3af-b8b4-44f6-a49f-9305791700f4')
+      .set('authorization', 'eeeeeeeeeeeeeee')
+      .end((err, res) => {
+        expect(res.status).to.equal(403);
+        expect(res.body.errors.body[0]).to.be.equal('Forbidden');
+        done();
+      });
+  });
+  it('It should return a 401 error for an empty token', done => {
+    chai
+      .request(app)
+      .post('/api/v1/bookmarks/7139d3af-b8b4-44f6-a49f-9305791700f4')
+      .set('authorization', '')
+      .end((err, res) => {
+        expect(res.status).to.equal(401);
+        expect(res.body.errors.body[0]).to.be.equal('You are not authorized');
+        done();
+      });
+  });
+
+  it('should have correct bookmark counts', done => {
+    chai
+      .request(app)
+      .get('/api/v1/article/7139d3af-b8b4-44f6-a49f-9305791700f4')
+      .set('Authorization', userToken)
+      .end((err, res) => {
+        expect(res).to.have.status(200);
+        const { bookmarkcount } = res.body.article;
+        expect(res.body.article).to.be.a('object');
+        expect(bookmarkcount).to.be.equal(54);
+        done();
+      });
+  });
+});
```

Comment:
You also need to test the count when a user removes a bookmark @vic3king

Reply:
@rotimi-babalola I tried to assert that the counts remains the same after adding and removing.
.But I'll break it up into two different assertions now.